src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -39,8 +39,110 @@ WRITE_CLASS_ENCODER(PeerImageId);
 
 typedef std::set<PeerImageId> PeerImageIds;
 
+enum NotifyOp {
+  NOTIFY_OP_IMAGE_ACQUIRE  = 0,
+  NOTIFY_OP_IMAGE_RELEASE  = 1,
+};
+
+struct ImageAcquirePayload {
+  static const NotifyOp NOTIFY_OP = NOTIFY_OP_IMAGE_ACQUIRE;
+
+  uint64_t request_id;
+  std::string global_image_id;
+  PeerImageIds peers;
```

Comment:
The peers actually come from "Mirror" when it scans the rbd_mirroring object and creates one Replayer per peer (eventually needs to be on Replayer per pool w/ multiple peers). Since an image will eventually be available on all peers, it doesn't really matter which specific peer has the image (or had the image removed) so long as we get notifications to re-verify a given image so that we can determine if we should remove it locally or if we should register ourselves in the journal of the peer.